File: tests/benchmark/corpus/go/safe/safe_interprocedural.go
package main

import (
	"net/http"
	"os"
	"path/filepath"
)

func sanitizePath(p string) string {
	return filepath.Clean(p)
}

func handler(w http.ResponseWriter, r *http.Request) {
	if !is_authenticated(r) {
		http.Error(w, "unauthorized", 401)
		return
	}
	path := r.URL.Query().Get("path")
	data, _ := os.ReadFile(sanitizePath(path))
	w.Write(data)
}

func is_authenticated(r *http.Request) bool {
	return r.Header.Get("Authorization") != ""
}

File: tests/benchmark/corpus/rust/safe/safe_interprocedural.rs
use std::env;
use std::fs;

fn canonicalize_path(p: &str) -> String {
    sanitize_input(p)
}

fn sanitize_input(s: &str) -> String {
    s.replace("..", "").replace("/", "_")
}

fn main() {
    let path = env::var("FILE_PATH").unwrap();
    let clean = canonicalize_path(&path);
    let contents = fs::read_to_string(&clean).unwrap();
    println!("{}", contents);
}
